plugins/base/src/main/kotlin/transformers/documentables/DefaultDocumentableMerger.kt

```diff
@@ -43,7 +45,16 @@ private fun <T> mergeExpectActual(
 
     fun analyzeExpectActual(sameDriElements: List<T>) = sameDriElements.reduce(reducer)
 
-    return elements.groupBy { it.dri }.values.map(::analyzeExpectActual)
+    fun T.isExpectActual() : Boolean =
+        this.safeAs<WithExtraProperties<T>>().let { it != null && it.extra[IsExpectActual] != null }
+
+    return elements.groupBy {
+        it.isExpectActual()
```

Comment:
Imho it won't work because of homo-hetero problem, we need to know if they are related in the expect-actual. This will merge cases when 2 classes are actual to different expect.

Reply:
i think when you define 2 expects in non-dependent sourcesets and then you have sourceset which depends on both of them, you cannot create actual because of name-clash, compiler won't let it go (how would he know to which expect you declare actual)

Reply:
Okey, I got it. You can have like: JSCommon -> JS, and JVMCommon -> JVM, and in this situation you can have expect/actual with the same name which you don't want to merge.